cwms-data-api/src/main/java/cwms/cda/data/dto/TimeSeries.java

```diff
@@ -240,6 +259,16 @@ private List<Column> getColumnDescriptor() {
                                 + "placeholder which can be important in irregular and psuedo regular timeseries."
             )
     )
+
+    // This class is used to deserialize the time-series data JSON into an object
+    // Solves the issue of the deserializer getting stuck in a loop
+    // and throwing a StackOverflowError when trying to handle the Record class directly
```

Comment:
This seems sketchy to me, why is your custom deserializer causing this?

Reply:
Removed custom deserializer